tests/FSharp.Test.Utilities/XunitSetup.fs

```diff
@@ -0,0 +1,12 @@
+namespace FSharp.Test
+
+open Xunit
+
+/// Exclude from parallelization. Execute test cases in sequence and do not run any other collections at the same time.
+[<CollectionDefinition(nameof DoNotRunInParallel, DisableParallelization = true)>]
+type DoNotRunInParallel = class end
```

Comment:
But so can we try using DisableParallelization in places where RunInSequence is used? What effect would it have?

Reply:
It would make things slower because DisableParallelization runs the excluded tests one by one after all the parallel collections are done. RunInSequence makes the marked class / module into normal xUnit implied collection i.e. it does not prevent it from running in parallel with other collections, it just makes the test cases and theory cases it contains run one after another, while other test collections can execute at the same time.

It's a bit complicated but it all works around the unfortunate design decision in xUnit that a test collection is the smallest unit of parallelization. The general idea with RunInSequence is to have as little of those as possible and work to eliminate them as the attribute marks tests that are not fully isolated from each other.

Maybe I rename the first one to DisableParallelization or DisableXunitParallelization?

Reply:
So... let's assume we have two test collections, TestCollection1 with tests TestA, TestB and TestC - and TestCollection2 with tests TestX, TestY, TestZ. With each of these attributes, which will be the order of execution?

Feel free to give some other example where the difference will be clear ;) 
I am just trying to get my head around it - if that's annoying, feel free to say "hey just play with this locally, you'll see the difference" :D 

If we decide to keep all these elements, it would be nice to add some guidance and explanations to the TestGuide.md. But that can be done right before the merge.